rpm.py

```diff
@@ -0,0 +1,853 @@
+    # gnu_architectures are those recognized by the canonical config.sub / config.guess
+    # and crosstool-ng. They are returned from ${CC} -dumpmachine and are a part of the
+    # sysroot.
+    gnu_architectures = dict({"ppc64le": "powerpc64le"})
+    try:
+        base_architecture = base_architectures[architecture]
+    except Exception:
+        base_architecture = architecture
+    try:
+        gnu_architecture = gnu_architectures[architecture]
+    except Exception:
+        gnu_architecture = architecture
+    architecture_bits = dict(
+        {
+            "architecture": architecture,
+            "base_architecture": base_architecture,
+            "gnu_architecture": gnu_architecture,
+            "bits": bits,
+        }
+    )
+    cdt = dict()
+    for k, v in iteritems(CDTs[cdt_name]):
+        if isinstance(v, string_types):
+            cdt[k] = v.format(**architecture_bits)
+        else:
+            cdt[k] = v
+
+    # Add undeclared dependencies. These can be baked into the global
+    # CDTs dict, passed in on the commandline or a mixture of both.
+    if "dependency_add" not in cdt:
+        cdt["dependency_add"] = dict()
+    if dependency_add:
+        for package_and_missed_deps in dependency_add:
+            as_list = package_and_missed_deps[0].split(",")
+            if as_list[0] in cdt["dependency_add"]:
+                cdt["dependency_add"][as_list[0]].extend(as_list[1:])
+            else:
+                cdt["dependency_add"][as_list[0]] = as_list[1:]
+
+    repomd_url = cdt["repomd_url"]
+    repo_primary = get_repo_dict(
+        repomd_url, "primary", massage_primary, cdt, config.src_cache
+    )
+    for package in packages:
+        write_conda_recipes(
+            recursive,
+            repo_primary,
+            package,
+            [architecture, "noarch"],
+            cdt,
+            output_dir,
+            override_arch,
+            config.src_cache,
+            build_number,
+        )
+
+
+@click.command()
+@click.argument("packages", nargs=-1, required=True)
+@click.option("--output-dir", default=".", type=str)
+@click.option("--version", default=None, type=str)
+@click.option("--recursive", default=False, is_flag=True)
+@click.option("--architecture", default=default_architecture, type=str)
+@click.option("--no-override-arch", default=False, is_flag=True)
+@click.option("--distro", default=default_distro, type=str)
+@click.option("--conda-forge-style", default=False, is_flag=True)
+@click.option("--single-sysroot", default=False, is_flag=True)
+@click.option("--build-number", default="2", type=str)
+def skeletonize(
+    packages,
+    output_dir,
+    version,
+    recursive,
+    architecture,
+    no_override_arch,
+    distro,
+    conda_forge_style,
+    single_sysroot,
+    build_number,
+):
+    global CONDA_FORGE_STYLE
+    CONDA_FORGE_STYLE = conda_forge_style
+
+    global SINGLE_SYSROOT
+    SINGLE_SYSROOT = single_sysroot
+
+    global CDTs
+    CDTs = _gen_cdts()
```

Comment:
Would it be possible to avoid globals? This might help make the script more robust since we are running it in a variety of contexts.

Reply:
Yeah, this needs fixing for sure. I'll def change that before I merge this. I was on the fence but you are right.

Reply:
fixed!